ctru-sys/src/bindings.rs

```diff
@@ -4736,27 +4772,119 @@ impl Default for pthread_attr_t {
         }
     }
 }
-pub type pthread_mutex_t = __uint32_t;
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub struct pthread_mutex_t {
+    pub type_: ::libc::c_int,
+    pub __bindgen_anon_1: pthread_mutex_t__bindgen_ty_1,
+}
+#[repr(C)]
+#[derive(Copy, Clone)]
+pub union pthread_mutex_t__bindgen_ty_1 {
+    pub normal: _LOCK_T,
+    pub recursive: _LOCK_RECURSIVE_T,
+}
+impl Default for pthread_mutex_t__bindgen_ty_1 {
+    fn default() -> Self {
+        let mut s = ::core::mem::MaybeUninit::<Self>::uninit();
+        unsafe {
+            ::core::ptr::write_bytes(s.as_mut_ptr(), 0, 1);
+            s.assume_init()
+        }
+    }
+}
+impl Default for pthread_mutex_t {
+    fn default() -> Self {
+        let mut s = ::core::mem::MaybeUninit::<Self>::uninit();
+        unsafe {
+            ::core::ptr::write_bytes(s.as_mut_ptr(), 0, 1);
+            s.assume_init()
+        }
+    }
+}
 #[repr(C)]
 #[derive(Debug, Default, Copy, Clone)]
 pub struct pthread_mutexattr_t {
-    pub is_initialized: ::libc::c_int,
-    pub recursive: ::libc::c_int,
+    pub type_: ::libc::c_int,
+}
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct pthread_cond_t {
+    pub clock_id: clockid_t,
+    pub cond: _COND_T,
 }
-pub type pthread_cond_t = __uint32_t;
 #[repr(C)]
 #[derive(Debug, Default, Copy, Clone)]
 pub struct pthread_condattr_t {
-    pub is_initialized: ::libc::c_int,
-    pub clock: clock_t,
+    pub clock_id: clockid_t,
 }
 pub type pthread_key_t = __uint32_t;
 #[repr(C)]
 #[derive(Debug, Default, Copy, Clone)]
 pub struct pthread_once_t {
-    pub is_initialized: ::libc::c_int,
-    pub init_executed: ::libc::c_int,
+    pub status: ::libc::c_int,
+}
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct pthread_barrier_t {
+    pub lock: _LOCK_T,
+    pub cond: _COND_T,
+    pub reload: ::libc::c_uint,
+    pub counter: ::libc::c_uint,
+    pub cycle: ::libc::c_uint,
+}
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct pthread_barrierattr_t {}
+#[repr(C)]
+#[derive(Debug, Default, Copy, Clone)]
+pub struct pthread_rwlock_t {
+    pub lock: _LOCK_T,
+    pub cond_r: _COND_T,
+    pub cond_w: _COND_T,
+    pub _bitfield_align_1: [u32; 0],
+    pub _bitfield_1: __BindgenBitfieldUnit<[u8; 4usize]>,
```

Comment:
huh, lots of new `pthread` related changes in these bindings... I wonder if this will break / can replace some of our `pthread-3ds` implementation...

Reply:
Looking into it more... we might be better off not upgrading `libctru`/`newlib` just yet: https://github.com/rust3ds/pthread-3ds/issues/26